src/main/typescript/node_modules/@atomist/rug/ast/TreeDiff.ts

```diff
@@ -0,0 +1,19 @@
+
+import { TextTreeNode } from "../tree/PathExpression"
+
+export function structurallyEquivalent(a: TextTreeNode, b: TextTreeNode): boolean {
+    if (a.children().length != b.children().length)
+        return false;
+
+    if (a.value() === b.value())
+        return true;
+
+    for (let i = 0; i < a.children().length; i++) {
+        const ax = a.children()[i] as TextTreeNode;
+        const bx = b.children()[i] as TextTreeNode;
+        if (!structurallyEquivalent(ax, bx)) {
+            console.log(`${ax} is not equivalent to ${bx}`);
```

Comment:
Good point. There are tests for this in `rugs` but it's temporarily in `rug` to help with solidifying AST stuff.